tests/Unit/NumericalAlgorithms/LinearOperators/Test_WeakDivergence.cpp

```diff
@@ -0,0 +1,163 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <limits>
+#include <random>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.hpp"
+#include "NumericalAlgorithms/LinearOperators/WeakDivergence.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+struct Var1 : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct Var2 : db::SimpleTag {
+  using type = tnsr::i<DataVector, Dim>;
+};
+
+template <size_t Dim>
+void test_weak_divergence_random_jacobian(const Mesh<Dim>& mesh) {
```

Comment:
So I've thought more about this since we talked yesterday.

I'm concerned that this is not a useful test: by using a constant flux, a non-constant jacobian, and checking the divergence is ~ zero ... is this not functionally the same as the test of the metric-identity jacobian?

Have I missed an important difference?

Perhaps an alternative would be to generate a random flux tensor, compute its divergence using the strong and weak expressions, then test those match at the correct (GL: interior; G: central) points?

Reply:
The reason for the current test is:
- verify that the jacobians are accounted for correctly
- that a consistent differentiation matrix is used in the metric identity jac computation and the div computation
I think checking this before check that the result matches what we expect (e.g. doing what you suggested) is useful because instead of just knowing "something went wrong" it's narrowed down.

I can add the test you suggested. I agree that it's useful